src/api/procedures/utils.ts

```diff
@@ -322,3 +330,166 @@ export function assertRequirementsNotTooComplex(
+  context: Context,
+  authRequest: AuthorizationRequest
+): Promise<void> {
+  if (authRequest.isExpired()) {
+    throw new PolymeshError({
+      code: ErrorCode.UnmetPrerequisite,
+      message: 'The Authorization Request has expired',
+      data: {
+        expiry: authRequest.expiry,
+      },
+    });
+  }
+  switch (authRequest.data.type) {
+    case AuthorizationType.RotatePrimaryKey:
+      return assertPrimaryKeyRotationAuthorizationValid(authRequest);
+    case AuthorizationType.AttestPrimaryKeyRotation:
+      return assertAttestPrimaryKeyAuthorizationValid(authRequest);
+    case AuthorizationType.TransferTicker:
+      return assertTransferTickerAuthorizationValid(context, authRequest.data);
+    case AuthorizationType.TransferAssetOwnership:
+      return assertTransferAssetOwnershipAuthorizationValid(context, authRequest.data);
+    case AuthorizationType.BecomeAgent:
+      // no additional checks
+      return;
+    case AuthorizationType.AddMultiSigSigner:
+      // no additional checks
+      return;
+    case AuthorizationType.PortfolioCustody:
+      // no additional checks
+      return;
+    case AuthorizationType.JoinIdentity:
+      return assertJoinIdentityAuthorizationValid(authRequest);
+    case AuthorizationType.AddRelayerPayingKey:
+      return assertAddRelayerPayingKeyAuthorizationValid(authRequest.data);
+    default:
+      throw new UnreachableCaseError(authRequest.data); // ensures switch statement covers all values
+  }
+}
+
+/**
+ * asserts valid  primary key rotation authorization
+ */
+export async function assertPrimaryKeyRotationAuthorizationValid(
+  authRequest: AuthorizationRequest
+): Promise<void> {
+  if (authRequest.target instanceof Identity) {
+    throw new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message: 'An Identity can not join another Identity',
+    });
+  }
+}
+
+/**
+ * asserts valid attest primary key authorization
+ */
+export async function assertAttestPrimaryKeyAuthorizationValid(
+  authRequest: AuthorizationRequest
+): Promise<void> {
+  if (!(await authRequest.issuer.isCddProvider())) {
+    throw new PolymeshError({
+      code: ErrorCode.UnmetPrerequisite,
+      message: 'Issuer must be a CDD provider',
+    });
+  }
+}
+
+/** asserts transfer ticker authorization is valid */
+export async function assertTransferTickerAuthorizationValid(
+  context: Context,
+  data: GenericAuthorizationData
+): Promise<void> {
+  const reservation = new TickerReservation({ ticker: data.value }, context);
+  const { status } = await reservation.details();
+  if (status === TickerReservationStatus.Free) {
+    throw new PolymeshError({
+      code: ErrorCode.UnmetPrerequisite,
+      message: 'The Ticker is not reserved',
+    });
+  }
+  if (status === TickerReservationStatus.TokenCreated) {
+    throw new PolymeshError({
+      code: ErrorCode.UnmetPrerequisite,
+      message: 'The ticker has already been used to create an Asset',
```

Comment:
Ticker should be uppercase here (we're not being very consistent accross the codebase though)